utilities/transactions/transaction_util.cc

```diff
@@ -104,8 +104,10 @@ Status TransactionUtil::CheckKey(DBImpl* db_impl, SuperVersion* sv,
     SequenceNumber seq = kMaxSequenceNumber;
     bool found_record_for_key = false;
 
-    Status s = db_impl->GetLatestSequenceForKey(sv, key, !need_to_read_sst,
-                                                &seq, &found_record_for_key);
+    Status s = db_impl->GetLatestSequenceForKey(
+        sv, key, !need_to_read_sst,
+        (min_uncommitted == kMaxSequenceNumber) ? snap_seq : min_uncommitted,
```

Comment:
It took me some time to remember why using min_committed as lower bound is correct here. Can we clarify that by: i) using a variable lower_bound and assign it before invoking GetLatestSequenceForKey. ii) add this inline comment on top of the variable:
```
// Sequence numbers <= snap_seq are not helpful since to check for conflicts we only care for the ones larger than snap_seq. When min_uncommitted is provided it guarantees that all the sequence numbers lower than min_uncommitted are already committed with a sequence number <= the snapshot, thus visible to it. Therefore any sequence < min_uncommitted are not helpful in detecting write conflict.
```